js/main.js

```diff
@@ -73,5 +80,54 @@ southLakeUnion.addRow(donutTable);
 wedgewood.addRow(donutTable);
 ballard.addRow(donutTable);
 
+// input form and button
+
+var donuts = document.getElementById('donuts');
+var donutForm = document.getElementById('donut-form');
+var donutButton = document.getElementById('donut-button');
+var donutData = [downtown, capitolHill, southLakeUnion, wedgewood, ballard];
   
+// button tag that saves location in following manner
+// instantiate a new DonutShop object with user supplied input tags
+// append a row to the table using the new object's hourly and daily
+// using the same HTML elements, update existing location tip: use an array
+// to hold DonutShop objects
+
+var handleDonutSubmit = function(event) { // function that handles the event
```

Comment:
This whole function looks to be functional, though you can simplify this quite a bit. If you have the ambition, and would like the practice/review, take a shot at refactoring this with your newer skills. I bet you would surprise yourself by being able to cut out 10-ish lines of code.